src/metabase/pulse.clj

```diff
@@ -414,13 +414,21 @@
   [{pulse-id :id, pulse-name :name, dashboard-id :dashboard_id, :as pulse} parts {:keys [recipients]}]
   (log/debug (u/format-color 'cyan (trs "Sending Pulse ({0}: {1}) with {2} Cards via email"
                                         pulse-id (pr-str pulse-name) (parts->cards-count parts))))
-  (let [email-recipients (filterv u/email? (map :email recipients))
-        timezone         (->> parts (some :card) defaulted-timezone)
-        dashboard        (update (t2/select-one Dashboard :id dashboard-id) :description markdown/process-markdown :html)]
-    {:subject      (subject pulse)
-     :recipients   email-recipients
-     :message-type :attachments
-     :message      (messages/render-pulse-email timezone pulse dashboard parts)}))
+  (let [user-recipients     (filter (fn [recipient] (and (u/email? (:email recipient))
+                                                         (some? (:id recipient)))) recipients)
+        non-user-recipients (filter (fn [recipient] (and (u/email? (:email recipient))
+                                                         (nil? (:id recipient)))) recipients)
+        timezone            (->> parts (some :card) defaulted-timezone)
+        dashboard           (update (t2/select-one Dashboard :id dashboard-id) :description markdown/process-markdown :html)
+        email-to-users      [{:subject      (subject pulse)
+                              :recipients   (mapv :email user-recipients)
+                              :message-type :attachments
+                              :message      (messages/render-pulse-email timezone pulse dashboard parts nil)}]]
```

Comment:
Now that we have precedent for this, could you actually change this code to send a separate email to each user as well? I think currently you can see all the other Metabase user email addresses when you receive it because it's all sent as a single email. I don't know if anyone's complained about this, but it'd be nice to fix especially in light of https://github.com/metabase/metabase/issues/10795